src/scheduler/server_info.c

```diff
@@ -1700,14 +1698,16 @@ free_server(server_info *sinfo)
 	if (sinfo == NULL)
 		return;
 
+	free_server_info(sinfo);
```

Comment:
Can you please write a comment explaining this move. It will help other developers to pay attention to order of operation.